Cannot build native ext on MSVC

```
(conanbuildenv) C:\Users\julien\Software\Others\openstudio_gems\sqlite3-ruby>ruby setup.rb config
---> lib
---> lib/sqlite3
<--- lib/sqlite3
<--- lib
---> ext
---> ext/sqlite3
C:/src/os-ruby/Ruby-prefix/src/Ruby-install/bin/ruby C:/Users/julien/Software/Others/openstudio_gems/sqlite3-ruby/ext/sqlite3/extconf.rb
checking for sqlite3.h... yes
checking for pthread_create() in pthread.lib... no
checking for dl.lib... no
checking for sqlite3_libversion_number() in sqlite3.lib... yes
checking for rb_proc_arity()... yes
checking for rb_integer_pack()... yes
checking for sqlite3_initialize()... yes
checking for sqlite3_backup_init()... yes
checking for sqlite3_column_database_name()... yes
checking for sqlite3_enable_load_extension()... yes
checking for sqlite3_load_extension()... yes
checking for sqlite3_open_v2()... yes
checking for sqlite3_prepare_v2()... yes
checking for sqlite3_int64 in sqlite3.h... yes
checking for sqlite3_uint64 in sqlite3.h... yes
creating Makefile
<--- ext/sqlite3
<--- ext

(conanbuildenv) C:\Users\julien\Software\Others\openstudio_gems\sqlite3-ruby>ruby setup.rb setup
---> lib
---> lib/sqlite3
<--- lib/sqlite3
<--- lib
---> ext
---> ext/sqlite3
nmake

Microsoft (R) Program Maintenance Utility Version 14.25.28611.0
Copyright (C) Microsoft Corporation.  All rights reserved.

generating sqlite3_native-x64-mswin64_140.def
compiling aggregator.c
aggregator.c
compiling backup.c
backup.c
compiling database.c
database.c
database.c(38): warning C4101: 'flags': unreferenced local variable
database.c(726): error C4028: formal parameter 1 different from declaration
database.c(726): warning C4022: 'sqlite3_exec': pointer mismatch for actual parameter 4
database.c(728): error C4028: formal parameter 1 different from declaration
database.c(728): warning C4022: 'sqlite3_exec': pointer mismatch for actual parameter 4
NMAKE : fatal error U1077: '"C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.25.28610\bin\HostX64\x64\cl.EXE"' : return code '0x2'
Stop.
'system nmake ' failed
Try 'ruby setup.rb --help' for detailed usage.
```

https://github.com/sparklemotion/sqlite3-ruby/blob/6fc5073ecc9c90d7458a43d726e92201e4c3ff65/ext/sqlite3/database.c#L725-L729

I don't really get why it throws it, since `ctx->db` is a `sqlite3*` which is what the `sqlite_exec` appears to be expecting.

From sqlite3.h, v 3.31.1:

```C
SQLITE_API int sqlite3_exec(
  sqlite3*,                                  /* An open database */
  const char *sql,                           /* SQL to be evaluated */
  int (*callback)(void*,int,char**,char**),  /* Callback function */
  void *,                                    /* 1st argument to callback */
  char **errmsg                              /* Error msg written here */
);
```

--- a/ext/sqlite3/database.c
+++ b/ext/sqlite3/database.c
@@ -1,6 +1,11 @@
 #include <sqlite3_ruby.h>
 #include <aggregator.h>
 
+#if _MSC_VER
+#pragma warning( push )
+#pragma warning( disable : 4028 )
+#endif
+
 #define REQUIRE_OPEN_DB(_ctxt) \
   if(!_ctxt->db) \
     rb_raise(rb_path2class("SQLite3::Exception"), "cannot use a closed database");
@@ -825,3 +830,7 @@ void init_sqlite3_database()
 
   rb_sqlite3_aggregator_init();
 }
+
+#if _MSC_VER
+#pragma warning( pop )
+#endif

